activerecord/test/cases/adapters/mysql2/mysql2_adapter_test.rb

```diff
@@ -322,6 +324,94 @@ def test_database_timezone_changes_synced_to_connection
     end
   end
 
+  def test_ignores_warnings_when_behaviour_ignore
+    old_action = ActiveRecord.db_warnings_action
+    ActiveRecord.db_warnings_action = :ignore
+    @conn.execute('SELECT 1 + "foo"')
+  ensure
+    ActiveRecord.db_warnings_action = old_action
+  end
+
+  def test_logs_warnings_when_behaviour_log
+    old_action = ActiveRecord.db_warnings_action
+    ActiveRecord.db_warnings_action = :log
+
+    mysql_warning = "[ActiveRecord::SqlWarning] Truncated incorrect DOUBLE value: 'foo' (1292)"
```

Comment:
This is a totally nitpicky thing that we don't enforce but since I'm requesting other changes I prefer when there's a new line before the assertions. Makes tests easier to read.